tensorflow/python/ops/math_ops.py

```diff
@@ -4857,6 +4857,68 @@ def sparse_segment_sum(
         name=name,
     )
 
+@tf_export("sparse.sampled_addmm")
+def sampled_addmm(indices,
+    values,
+    dense_shape,
+    mat1,
+    mat2,
+    beta=1.0,
+    alpha=1.0
+):
+  sparse_rank = len(dense_shape)
+  sparse_num_batches = dense_shape[0] if sparse_rank == 3 else 1
+  sparse_num_rows = dense_shape[0 if sparse_rank == 2 else 1]
+  sparse_num_cols = dense_shape[1 if sparse_rank == 2 else 2]
+
+  values_shape = values.get_shape()
+  values_shape_list = values_shape.as_list()
+  indices_shape = indices.get_shape()
+  indices_shape_list = indices_shape.as_list()
+
+  mat1_shape = mat1.get_shape()
+  mat1_shape_list = mat1_shape.as_list()
+  mat2_shape = mat2.get_shape()
+  mat_rank = mat1_shape.ndims
+
+  mat_num_batches = mat1_shape_list[0] if mat_rank == 3 else 1
+  mat1_num_rows = mat1_shape_list[0 if mat_rank == 2 else 1]
+  mat2_num_cols = mat2_shape.as_list()[1 if mat_rank == 2 else 2]
+
+  if values_shape.ndims != sparse_rank - 1:
```

Comment:
These shape validation checks are probably best done by the kernel itself, or its shape function when registering the op (if ranks are known).  You could also let the kernel itself determine the number of batch dimensions, etc.

The python code in TF is mainly used to set up a graph of execution, and may not have full tensor size information at the time (e.g. if the compiled graph is meant to work for tensors of varying sizes).  But by the time the graph is executed and we enter the kernel, the full size of the inputs will be known.